engine/src/main/java/org/terasology/input/cameraTarget/TargetSystem.java

```diff
@@ -44,6 +44,15 @@ public TargetSystem(BlockEntityRegistry blockRegistry, Physics physics) {
         this.physics = physics;
     }
 
+    /**
+     * Gets the position of the block that is currently targeted. If there is currently no target, this will return
+     * a null reference (isTargetAvailable() would have returned false in this case).
+     * @return the target block position in world coordinates, a vector of 3 integers.
+     */
+    public Vector3i getTargetBlockPosition() {
+        return targetBlockPos;
+    }
+
```

Comment:
Are you referring to this?

```
LocationComponent location = target.getComponent(LocationComponent.class);
        if (location != null && targetBlockPos != null) {
            location.setLocalPosition(targetBlockPos.toVector3f());
        }
```
As I understand this code, the `LocationComponent` is a reference to the one stored within the entity. So by calling `setLocalPosition()` on that block, it mutates the location component on the actual entity. I'm not sure what happens downstream of / surrounding this code. Once again, the Law of Least Intervention is a decent one to follow on large code bases being worked on by many different people.